tests/functional/listing/test.js

```diff
@@ -0,0 +1,476 @@
+    }
+} else {
+    const ref = require('../../../conf/authdata.json');
+    AWS.config.update({
+        accessKeyId: ref.accounts[0].keys[0].access,
+        secretAccessKey: ref.accounts[0].keys[0].secret,
+        region: 'us-west-1',
+        sslEnabled: false,
+        endpoint: 'http://localhost:8000',
+        s3ForcePathStyle: true,
+        apiVersions: { s3: '2006-03-01' },
+        signatureVersion: 'v4',
+        signatureCache: false,
+    });
+}
+
+const s3 = new AWS.S3();
+
+describe('Listing corner cases tests', () => {
+    before(done => {
+        s3.createBucket(
+            { Bucket }, (err, data) => {
+                if (err) {
+                    done(err, data);
+                }
+                async.eachLimit(
+                    objects, 999, (o, next) => {
+                        s3.putObject(o, (err, data) => {
+                            next(err, data);
+                        });
+                    }, done);
+            });
+    });
+    after(done => {
+        s3.listObjects({ Bucket }, (err, data) => {
+            async.eachLimit(data.Contents, 999, (o, next) => {
+                s3.deleteObject({ Bucket, Key: o.Key }, next);
+            }, () => {
+                s3.deleteBucket({ Bucket }, done);
+            });
+        });
+    });
+    it('should list everything', done => {
+        s3.listObjects({ Bucket }, (err, data) => {
+            assert.strictEqual(err, null);
+            cutAttributes(data);
+            assert.deepStrictEqual(data, {
+                IsTruncated: false,
+                Marker: '',
+                Contents: [
+                    objects[0].Key,
+                    objects[1].Key,
+                    objects[2].Key,
+                    objects[3].Key,
+                    objects[4].Key,
+                    objects[5].Key,
+                    objects[6].Key,
+                    objects[7].Key,
+                    objects[8].Key,
+                    objects[9].Key,
+                ],
+                Name: Bucket,
+                Prefix: '',
+                MaxKeys: 1000,
+                CommonPrefixes: [],
+            });
+            done();
+        });
+    });
+    it('should list with valid marker', done => {
+        s3.listObjects(
+            { Bucket,
+              Delimiter: '/',
+              Marker: 'notes/summer/1.txt',
+            },
+            (err, data) => {
+                assert.strictEqual(err, null);
+                cutAttributes(data);
+                assert.deepStrictEqual(data, {
+                    IsTruncated: false,
+                    Marker: 'notes/summer/1.txt',
+                    Contents: [],
+                    Name: Bucket,
+                    Prefix: '',
+                    Delimiter: '/',
+                    MaxKeys: 1000,
+                    CommonPrefixes: [],
+                });
+                done();
+            });
+    });
+    it('shoudle list with bad marker', done => {
```

Comment:
because S3 this marker is not expected regarding the content of the bucket, rewording.